docs/questions/query-builder/expressions/regexextract.md

```diff
@@ -0,0 +1,140 @@
+---
+title: Regexextract
+---
+
+# Regexextract
+
+`regexextract` uses [regular expressions (regex)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions) to get a specific part of your text.
+
+`regexextract` is ideal for text that has little to no structure, like URLs or freeform survey responses. If you're working with SKU numbers, IDs, or other types of codes, check out the simpler [substring](../expressions/substring.md) expression instead.
+
+Use this expression to create custom columns with shorter, more readable labels for things like:
+
+- filter dropdown menus, 
+- chart labels, or
+- embedding parameters.
+
+| Syntax                                                        | Example                                                                        |
+|---------------------------------------------------------------|--------------------------------------------------------------------------------|
+| `regexextract(text, regular_expression)`                      | `regexextract("Oh dear! Oh dear! I shall be too late!", "Oh dear(.*)Oh dear")`|
+| Gets a specific part of your text using a regular expression. | "!"                                                                            |
+
+## Searching and cleaning text
+
+Let's say that you have web data with a lot of different URLs, and you want to map each URL to a shorter, more readable campaign name.
+
+| URL                                                   | Campaign Name |
+|-------------------------------------------------------|---------------|
+| https://www.metabase.com/docs/?utm_campaign=alice     | alice         |
+| https://www.metabase.com/learn/?utm_campaign=neo      | neo           |
+| https://www.metabase.com/glossary/?utm_campaign=candy | candy         |
+
+You can create a custom column **Campaign Name** with the expression:
+
+```
+regexextract([URL], "^[^?#]+\?utm_campaign=(.*)")
```

Comment:
I'd double check to make sure this regex behaves as expected. I don't know if regexextract cares about groups, and may just return matches.